src/openssl.c

```diff
@@ -419,27 +420,20 @@ _libssh2_cipher_crypt(_libssh2_cipher_ctx * ctx,
 {
     unsigned char buf[EVP_MAX_BLOCK_LENGTH];
     int ret;
+    int outlen;
     (void) algo;
     (void) encrypt;
 
 #ifdef HAVE_OPAQUE_STRUCTS
-    ret = EVP_Cipher(*ctx, buf, block, blocksize);
+    ret = EVP_CipherUpdate(*ctx, buf, &outlen, block, blocksize);
```

Comment:
I _think_ simply switching to using `EVP_CipherUpdate` here clears up the return code complexity.